lib/signed-xml.js

```diff
@@ -359,35 +359,65 @@ SignedXml.defaultNsForPrefix = {
 
 SignedXml.findAncestorNs = findAncestorNs;
 
-SignedXml.prototype.checkSignature = function (xml, callback) {
-  if (callback != null && typeof callback !== "function") {
-    throw new Error("Last parameter must be a callback function");
+/**
+ * Builds the contents of a KeyInfo element as an XML string.
+ *
+ * For example, if the value of the prefix argument is 'foo', then
+ * the resultant XML string will be "<foo:X509Data></foo:X509Data>"
+ *
+ * @param cert the certificate as a string or array of strings (see https://www.w3.org/TR/2008/REC-xmldsig-core-20080610/#sec-X509Data)
+ * @param prefix an optional namespace alias to be used for the generated XML
+ * @return an XML string representation of the contents of a KeyInfo element
+ */
+SignedXml.getKeyInfoContent = function ({ publicCert = null, prefix = null } = {}) {
+  if (publicCert == null) {
+    return null;
   }
 
-  this.validationErrors = [];
-  this.signedXml = xml;
+  prefix = prefix ? prefix + ":" : prefix;
 
-  if (!this.keyInfoProvider) {
-    var err = new Error("cannot validate signature since no key info resolver was provided");
-    if (!callback) {
-      throw err;
-    } else {
-      callback(err);
-      return;
-    }
+  let x509Certs = "";
+  if (Buffer.isBuffer(publicCert)) {
+    publicCert = publicCert.toString("latin1");
   }
 
-  this.signingKey = this.keyInfoProvider.getKey(this.keyInfo);
-  if (!this.signingKey) {
-    var err2 = new Error("key info provider could not resolve key info " + this.keyInfo);
-    if (!callback) {
-      throw err2;
-    } else {
-      callback(err2);
-      return;
+  if (typeof publicCert === "string") {
+    publicCert = publicCert.match(utils.MULTI_PEM_SPLIT);
```

Comment:
We should make sure that private keys are excluded during this check.
This includes tags:

```
-----BEGIN PRIVATE KEY-----
-----BEGIN RSA PRIVATE KEY-----
-----BEGIN ENCRYPTED PRIVATE KEY-----
```

Basically any tags containing the word PRIVATE.

Reply:
Have you considered using the 'crypto' module to parse private and public keys?

Reply:
I feel like at some point we have to not worry about people doing the wrong thing with such clear labels, but I'll see what I can do. Want to write some tests for this?

Reply:
> I feel like at some point we have to not worry about people doing the wrong thing with such clear labels

At some point yes, but IMO private key confidentiality is not that point.

I will look at it tomorrow if I find the time.